processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -426,4 +518,37 @@ public Row apply(Object input)
       }
     };
   }
+
+
+  public static Set<String> optimizedExtractionDims(GroupByQuery query, final boolean getPre)
```

Comment:
The past tense here makes me feel like it is already optimized after this call or something.  Can we change it to "extractionsToRewrite" or something like that?